application-diagram-ui/src/main/resources/Diagram/DiagramSheet.xml

```diff
@@ -634,15 +634,32 @@ define('diagram-graph-xml-filter', ['jquery', 'diagram-config', 'diagram-link-ha
   };
 
   // Replace custom internal links with absolute URLs (but keep some metadata to indicate the entity reference).
+  // Consider both the cases when the node is UserObject or mxCell.
   var onExportLink = function(node) {
-    var link = $(node).attr('link');
+    var tagName = node.tagName.toLowerCase();
+    var value = $(node).attr('value');
+    if (value) {
+      var valueInsideTag = value.match(/&lt;a\shref=".*"&gt;/g)[0];
+    }
+    var link = tagName == 'userobject' ? $(node).attr('link') : $(valueInsideTag).attr('href');
     if (diagramLinkHandler.isXWikiCustomLink(link)) {
       var url = diagramLinkHandler.getURLFromCustomLink(link);
-      var absoluteURL = $('&lt;a/&gt;').attr('href', url).prop('href');
-      $(node).attr({
-        link: absoluteURL,
+      var absoluteURL = $('&lt;a/&gt;').attr('href', url);
+      var attributesMap = {
+        'link': absoluteURL.prop('href'),
         'data-link': link
-      });
+      };
+      if (tagName != 'userobject') {
+        // Rebuild the value attribute, that might not contain just the link, with the new url.
+        var absoluteURLValue = value
+          .replace(/(&lt;a\shref=")(data:.*)("&gt;)/g, "$1$absoluteURL$3")
+          .replace('$absoluteURL', absoluteURL[0]);
```

Comment:
I would use jQuery here also instead of a regular expression. You can probably keep a reference to the anchor element found above (when you get the value of ``href``) and update it. Then you can use ``$(...).html()`` to get the inner HTML to put in the ``mxCell`` value.